contracts/governance/extensions/GovernorCountingSimple.sol

```diff
@@ -15,7 +15,8 @@ abstract contract GovernorCountingSimple is Governor {
     enum VoteType {
         Against,
         For,
-        Abstain
+        Abstain,
+        Fractional
```

Comment:
I'm not convinced it's fine to extend the previous enum. I think it should be generally a bad practice to extend enums because it's assigning an arbitrary meaning to an unused `uint8` (in the context of `GovernorCountingSimple`).

This would make sense if `GovernorCountingFractional` extended from `GovernorCountingSimple` or a common implementation, but we've deliberately decided to keep their implementations separate so I think it may be better to have a vote type enum exclusive for `GovernorCountingFractional`.

Reply:
I think there is a value in having the enum declared only once.

If we declare 
```
// in GovernorCountingSimple
enum VoteType { Against, For, Abstain }
// in GovernorCountingFractional
enum VoteTypeFractional { Against, For, Abstain, Fractional }
```
then we run the risk of someone else, not aware of `GovernorCountingFractional`, extending `VoteType` with another 4th option that is not equivalent to Fractional. This would be terrible to indexers.

IMO the only good options are:
- declaring Fractional as a "core" enum entry. Ideally I would declare the enum elsewhere, but that would be a big breaking change
- not delcaring Franctional as an enum entry, and use `type(uint8).max` instead.

Reply:
Yeah, I think it's terrible for general-purpose indexers, but there is already a clarification in [the event](https://github.com/OpenZeppelin/openzeppelin-contracts/blob/93f460f8854bbfcdd0d1150c786b61ce7646330c/contracts/governance/IGovernor.sol#L145):

> `support` values should be seen as buckets. Their interpretation depends on the voting module used.

I don't think it's a big deal for indexers who're tracking their own Governor, since they will never have a conflict definition for `support`. And even if that's the case, I think there might be already other uses of `support` we're not aware of.

> not declaring Fractional as an enum entry, and use type(uint8).max instead.

This sounds like a good option. I don't particularly prefer one over the other.

Reply:
I implemented the `type(uint8).max` option.

I'd love @frangio 's feedback.

Reply:
> then we run the risk of someone else, not aware of `GovernorCountingFractional`, extending `VoteType` with another 4th option that is not equivalent to Fractional. This would be terrible to indexers.

The point of `COUNTING_MODE` was that contracts can declare how the `support` value should be interpreted. As long as `COUNTING_MODE` is different, the enums can be completely different.

IMO this contract should have a separate enum, preserving the order and adding the new `Fractional` variant, and declare something like `support=fractional` in `COUNTING_MODE`.

Reply:
> preserving the order and adding the new `Fractional` variant

Actually, I don't have a strong preference for this. It would be fine to use `type(uint8).max`, but this still requires changing the value of `support` in the counting mode!

Reply:
>but this still requires changing the value of support in the counting mode!

current commit uses:
```solidity
    function COUNTING_MODE() public pure virtual override returns (string memory) {
        return "support=bravo,params&quorum=for,abstain&params=fractional";
    }
```

do you feel that is ok? would you do `support=bravo+params` or something completelly different ?

Reply:
Yeah I just saw that. That's good, although it should be added to the documentation for `COUNTING_MODE` in `IGovernor`.

The comma is good.

Reply:
Sorry, second thoughts... I don't understand the rationale for `params`. I would've named it `support=bravo,fractional`.

Reply:
Is the idea that `,params` means a value of `type(uint8).max` defers the meaning of the support to the content of params? Maybe that's good. I'll wait to see the documentation.

Reply:
Rational comes from here: https://github.com/ScopeLift/flexible-voting/blob/e5de2efd1368387b840931f19f3c184c85842761/src/GovernorCountingFractional.sol#L55

Will check with tally if they use it, and if `support=bravo,fractional` would work for them

Reply:
I think `support=bravo,params` is okay though!

Reply:
Also agree with the current `COUNTING_MODE`
I'm solving since `type(uint8).max` option was used and we all seem aligned